storage/innobase/buf/buf0buf.cc

```diff
@@ -3777,6 +3777,16 @@ dberr_t buf_page_t::read_complete(const fil_node_t &node)
   if (err == DB_PAGE_CORRUPTED || err == DB_DECRYPTION_FAILED)
   {
 release_page:
+    if (node.space->full_crc32() && node.space->crypt_data
+        && recv_recovery_is_on())
+    {
+      /* Recover from doublewrite buffer */
+      err= recv_sys.dblwr.recover_encrypted_page(
+             node.space, id().page_no(),
+             const_cast<byte*>(read_frame));
```

Comment:
When this is not returning `DB_SUCCESS`, it is discarding the previous `err`, which could be `DB_PAGE_CORRUPTED` or `DB_DECRYPTION_FAILED`. Maybe the function call should just return `bool` and we should preserve the current `err` across the call?

Also, for an accurate file name in `sql_print_information` reporting about successful recovery, we should pass `node` instead of `node.space` to the function. That would work for a multi-page system tablespace.